src/t8_cmesh/t8_cmesh_vertex_conn_tree_to_vertex.cxx

```diff
@@ -0,0 +1,166 @@
+   * We need to store the local indices as well, since they will not be
+   * sorted. Hence we need to sort later. */
+  using global_local_index_pair = std::pair<t8_gloidx_t, t8_locidx_t>;
+  std::vector<std::vector<global_local_index_pair>> global_ids_per_tree (num_local_trees_and_ghosts);
+
+  /* Iterate over all entries of the vertex to tree list. */
+  for (auto &[global_vertex, vertex_list] : vtt) {
+    /* Iterate over all entries of the vertex list,
+     * thus, each entry is a local tree id and vertex index. */
+    for (auto &[tree_index, vertex_index] : vertex_list) {
+      global_local_index_pair new_pair (global_vertex, vertex_index);
+      global_ids_per_tree[tree_index].push_back (new_pair);
+    }
+  }
+  /* Now sort the list and add the global indices of each tree. */
+  for (t8_locidx_t itree = 0; itree < num_local_trees_and_ghosts; ++itree) {
+    auto &tree_indices = global_ids_per_tree[itree];
+    /* Sort according to local index. */
+    std::sort (tree_indices.begin (), tree_indices.end (),
+               [] (global_local_index_pair &a, global_local_index_pair &b) { return a.second < b.second; });
+
+    /* Compute number of vertices of this tree. */
+    /* Get the trees class depending on whether it is a local tree or ghost. */
+    const t8_eclass_t tree_class = itree < num_local_trees
+                                     ? t8_cmesh_get_tree_class (cmesh_from, itree)
+                                     : t8_cmesh_get_ghost_class (cmesh_from, itree - num_local_trees);
+
+    const int num_tree_vertices = t8_eclass_num_vertices[tree_class];
+
+    /* Size of list must match number of tree vertices. */
+    SC_CHECK_ABORTF (
+      tree_indices.size () == (size_t) num_tree_vertices,
+      "ERROR. Number of mapped local tree vertices for tree %i does not equal number of tree vertices. %zu != %i\n",
+      itree, tree_indices.size (), num_tree_vertices);
+    /* We now build the array of global indices of this tree, sorted 
+     * by the trees local indices. */
+    t8_gloidx_t *global_tree_indices = T8_ALLOC (t8_gloidx_t, num_tree_vertices);
+    for (int ivertex = 0; ivertex < num_tree_vertices; ++ivertex) {
+      global_tree_indices[ivertex] = tree_indices[ivertex].first;
+    }
+    const t8_gloidx_t global_tree_id = t8_cmesh_get_global_id (cmesh_from, itree);
+    set_global_vertex_ids_of_tree_vertices (cmesh, global_tree_id, global_tree_indices, num_tree_vertices);
+    T8_FREE (global_tree_indices);
+  }
+}
+
+/** Set all global vertex ids of a local tree.
+  * \param [in] cmesh The considered cmesh
+  * \param [in] local_tree A local tree id of \a cmesh
+  * \param [in] global_vertex_id The ids of the global vertices in order of \a local_tree's vertices.
+  * \param [in] num_vertices Must match the number of vertices of \a local_tree
+  *
+  * \note Cmesh must not be committed.
+  */
+void
+t8_cmesh_vertex_conn_tree_to_vertex::set_global_vertex_ids_of_tree_vertices (const t8_cmesh_t cmesh,
```

Comment:
Could we template this function so that it takes every iterable container like `std::vector` or `std::array?`